message/validation/alan/common_checks.go

```diff
@@ -0,0 +1,91 @@
+package msgvalidation
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/attestantio/go-eth2-client/spec/phase0"
+	spectypes "github.com/bloxapp/ssv-spec/alan/types"
+	"golang.org/x/exp/slices"
+)
+
+// ownCommittee should be called only when WithOwnOperatorID is set
+func (mv *messageValidator) ownCommittee(committee []spectypes.OperatorID) bool {
+	return slices.ContainsFunc(committee, func(operatorID spectypes.OperatorID) bool {
+		return operatorID == mv.operatorDataStore.GetOperatorID()
+	})
+}
+
+func (mv *messageValidator) committeeRole(role spectypes.RunnerRole) bool {
+	return role == spectypes.RoleCommittee
+}
+
+func (mv *messageValidator) validateSlotTime(messageSlot phase0.Slot, role spectypes.RunnerRole, receivedAt time.Time) error {
+	if mv.earlyMessage(messageSlot, receivedAt) {
+		return ErrEarlyMessage
+	}
+
+	if lateness := mv.lateMessage(messageSlot, role, receivedAt); lateness > 0 {
+		e := ErrLateMessage
+		e.got = fmt.Sprintf("late by %v", lateness)
+		return e
+	}
+
+	return nil
+}
+
+func (mv *messageValidator) earlyMessage(slot phase0.Slot, receivedAt time.Time) bool {
+	return mv.netCfg.Beacon.GetSlotEndTime(mv.netCfg.Beacon.EstimatedSlotAtTime(receivedAt.Unix())).
+		Add(-clockErrorTolerance).Before(mv.netCfg.Beacon.GetSlotStartTime(slot))
+}
+
+func (mv *messageValidator) lateMessage(slot phase0.Slot, role spectypes.RunnerRole, receivedAt time.Time) time.Duration {
+	var ttl phase0.Slot
+	switch role {
+	case spectypes.RoleProposer, spectypes.RoleSyncCommitteeContribution:
+		ttl = 1 + lateSlotAllowance
+	case spectypes.RoleCommittee, spectypes.RoleAggregator:
+		ttl = 32 + lateSlotAllowance
+	default:
+		panic("unexpected role")
+	}
+
+	deadline := mv.netCfg.Beacon.GetSlotStartTime(slot + ttl).
+		Add(lateMessageMargin).Add(clockErrorTolerance)
+
+	return mv.netCfg.Beacon.GetSlotStartTime(mv.netCfg.Beacon.EstimatedSlotAtTime(receivedAt.Unix())).
+		Sub(deadline)
+}
+
+func (mv *messageValidator) validateDutyCount(
+	validatorIndices []phase0.ValidatorIndex,
+	state *SignerState,
+	msgID spectypes.MessageID,
+	newDutyInSameEpoch bool,
+) error {
+	var dutyLimit int
+
+	switch msgID.GetRoleType() {
+	case spectypes.RoleAggregator, spectypes.RoleValidatorRegistration, spectypes.RoleVoluntaryExit:
+		dutyLimit = 2
+
+	case spectypes.RoleCommittee:
+		dutyLimit = 2 * len(validatorIndices)
```

Comment:
For the RoleCommittee, the limit should be:
- 2 * V, if no validators in the committee are in a "sync committee epoch"
- 32, otherwise

Reply:
@MatheusFranco99 do I understand it correctly?
- $2 * V$ if there's a sync committee duty for at least one validator index in the message's epoch
- $32$ otherwise

Reply:
Actually, 2*V if no validators have a sync committee duty